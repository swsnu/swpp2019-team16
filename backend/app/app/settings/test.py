from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
