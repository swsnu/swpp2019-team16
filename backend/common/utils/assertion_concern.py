class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
