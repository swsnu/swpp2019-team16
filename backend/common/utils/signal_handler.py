import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
