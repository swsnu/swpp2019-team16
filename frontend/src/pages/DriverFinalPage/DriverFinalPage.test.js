import React from 'react';
import DriverFinalPage from './DriverFinalPage';
import { renderWithRedux } from '../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<DriverFinalPage />', () => {
  it('SHOULD match with snapshot', async () => {
    const state = {
      auth: {
        auth: {
          id: 1,
        },
      },
      user: {
        user: {
          id: 1,
          user: {
            id: 2,
            point: 50000,
          },
        },
      },
      group: {
        group: {
          gourpId: 1,
          cost: 5000,
        },
      },
    };

    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/driverfinal']}>
        <Route component={DriverFinalPage} path={'/driverfinal'} />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
