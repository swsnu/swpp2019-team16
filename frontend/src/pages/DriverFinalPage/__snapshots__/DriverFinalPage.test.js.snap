// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DriverFinalPage /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h4"
        >
          Ya-Ta!
        </h4>
        <div>
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-root-63"
    >
      <div
        class="makeStyles-whiteBox-65"
      >
        <div
          class="makeStyles-container-64"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Total Earning: ₩
            6000
          </h3>
        </div>
        <div
          class="makeStyles-container-64"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Current Point: ₩
            50000
          </h3>
        </div>
        <div
          class="makeStyles-container-64"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            You earned ₩
            1000
             more!
          </h3>
        </div>
        <div
          class="makeStyles-container-64"
        >
          <div
            class="makeStyles-button-66"
            fullwidth="false"
            variant="contained"
          >
            Go To Main
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
